AutomatedTesting/Gem/PythonTests/Terrain/EditorScripts/Terrain_SupportsPhysics.py

```diff
@@ -0,0 +1,164 @@
+    entity1_components_to_add = ["Axis Aligned Box Shape", "Terrain Layer Spawner", "Terrain Height Gradient List", "Terrain Physics Heightfield Collider", "PhysX Heightfield Collider"]
+    entity2_components_to_add = ["Vegetation Reference Shape", "Gradient Transform Modifier", "FastNoise Gradient"]
+    ball_components_to_add = ["Sphere Shape", "PhysX Collider", "PhysX Rigid Body"]
+    terrain_spawner_entity = hydra.Entity("TestEntity1")
+    terrain_spawner_entity.create_entity(azmath.Vector3(512.0, 512.0, 50.0), entity1_components_to_add)
+    Report.result(Tests.create_terrain_spawner_entity, terrain_spawner_entity.id.IsValid())
+    height_provider_entity = hydra.Entity("TestEntity2")
+    height_provider_entity.create_entity(azmath.Vector3(0.0, 0.0, 0.0), entity2_components_to_add,terrain_spawner_entity.id)
+    Report.result(Tests.create_height_provider_entity, height_provider_entity.id.IsValid())
+    # 2a) Create a ball at 600.0, 600.0, 46.0 - This position is bot too high over the heighfield so will collide in a reasonable time
+    ball = hydra.Entity("Ball")
+    ball.create_entity(azmath.Vector3(600.0, 600.0, 46.0), ball_components_to_add)
+    Report.result(Tests.create_test_ball, ball.id.IsValid())
+    # Give everything a chance to finish initializing.
+    general.idle_wait_frames(1)
+
+    # 3) Start the Tracer to catch any errors and warnings
+    with Tracer() as section_tracer:
+        # 4) Change the Axis Aligned Box Shape dimensions
+        box_dimensions = azmath.Vector3(SET_BOX_X_SIZE, SET_BOX_Y_SIZE, SET_BOX_Z_SIZE)
+        terrain_spawner_entity.get_set_test(0, "Axis Aligned Box Shape|Box Configuration|Dimensions", box_dimensions)
+        box_shape_dimensions = hydra.get_component_property_value(terrain_spawner_entity.components[0], "Axis Aligned Box Shape|Box Configuration|Dimensions")
+        Report.result(Tests.box_dimensions_changed, box_dimensions == box_shape_dimensions)
+        
+        # 5) Set the Vegetaion Shape reference to TestEntity1
+        height_provider_entity.get_set_test(0, "Configuration|Shape Entity Id", terrain_spawner_entity.id)
+        entityId = hydra.get_component_property_value(height_provider_entity.components[0], "Configuration|Shape Entity Id")
+        Report.result(Tests.shape_changed, entityId == terrain_spawner_entity.id)
+
+        # 6) Set the FastNoise Gradient frequency to 0.01
+        Frequency = 0.01
+        height_provider_entity.get_set_test(2, "Configuration|Frequency", Frequency)
+        FrequencyVal = hydra.get_component_property_value(height_provider_entity.components[2], "Configuration|Frequency")
+        Report.result(Tests.frequency_changed, math.isclose(Frequency, FrequencyVal, abs_tol = 0.00001))
+
+        # 7) Set the Gradient List to TestEntity2
+        pte = hydra.get_property_tree(terrain_spawner_entity.components[2])
```

Comment:
would it be possible to rename this variable to something more explanatory?

Reply:
Will fix.